src/frontend/web_application/src/scenes/Contact/components/PublicKeyForm/presenter.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Field } from 'redux-form';
+import { withI18n, Trans } from '@lingui/react';
+import renderReduxField from '../../../../services/renderReduxField';
+import { Button, Confirm, FieldErrors, Fieldset, FormColumn, FormGrid, FormRow, Icon, Legend, TextFieldGroup as TextFieldGroupBase, TextareaFieldGroup as TextareaFieldGroupBase } from '../../../../components';
+
+import './style.scss';
+
+const TextFieldGroup = renderReduxField(TextFieldGroupBase);
+const TextareaFieldGroup = renderReduxField(TextareaFieldGroupBase);
+
+@withI18n()
+class PublicKeyForm extends PureComponent {
+  static propTypes = {
+    errors: PropTypes.arrayOf(PropTypes.string),
+    i18n: PropTypes.shape({}).isRequired,
+    contactId: PropTypes.string.isRequired,
+    publicKey: PropTypes.shape({}),
+    handleSubmit: PropTypes.func.isRequired,
+    form: PropTypes.string.isRequired,
+    createPublicKey: PropTypes.func.isRequired,
+    updatePublicKey: PropTypes.func.isRequired,
+    deletePublicKey: PropTypes.func.isRequired,
+    onSuccess: PropTypes.func,
+    onCancel: PropTypes.func,
+  };
+
+  static defaultProps = {
+    publicKey: undefined,
+    onSuccess: undefined,
```

Comment:
usually I set a noop: `onSuccess: () => {},`

so you will not have to add a condition afterward: `if (onSuccess) onSuccess()`